Day3Part2.js

```diff
@@ -0,0 +1,53 @@
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+readline.question(`enter your data `, function LifeSupportRating(data) {
+  var Data = data.split(" ");
+  zeroCounter = 0;
+  oneCounter = 0;
+
+  let arro2 = [...Data];
+  let arrco2 = [...Data];

```

Comment:
🤔 These names make no sense to a reader. Also the variables are not reassigned so you should use `const` and they are **not** needed. You are not manipulating `dataArray` so you can directly use it.

You can use `getKeyRating(dataArray, 'some-key')` directly instead of with these variables.

If you thought that you are manipulating the array in `getKeyRating` you should read how the `filter` method works, because it returns a copy instead of changing the original array.